File: all-submissions/0008-string-to-integer-atoi/solution.java
class Solution {
    public int myAtoi(String s) {
        int answer = 0;
        int sign = 1;
        int numStart = 0;
        for(int i=0; i<s.length();i++) {
            if (s.charAt(i) != ' ') {
                numStart = i;
                break;
            }
        }
        if (numStart == s.length()) { return answer;}
        if (!Character.isDigit(s.charAt(numStart)) && s.charAt(numStart) != '-' && s.charAt(numStart) != '+') {
            return answer;
        }
        if(!Character.isDigit(s.charAt(numStart))){
            if (s.charAt(numStart) == '-'){
                sign = -1;
            } 
            numStart++;
        }
        for(int i=numStart; i<s.length(); i++){
            if (!Character.isDigit(s.charAt(i))){
                return answer*sign;
            }
            if (answer > Integer.MAX_VALUE/10 || (answer == Integer.MAX_VALUE/10 && (s.charAt(i) - '0') > Integer.MAX_VALUE%10)) {
                return sign == 1 ? Integer.MAX_VALUE : Integer.MIN_VALUE;
            }
            answer = answer*10+(s.charAt(i) - '0');
        }
        return answer * sign;


    }
}

File: all-submissions/0008-string-to-integer-atoi/solution.py
class Solution:
    def myAtoi(self, s: str) -> int:
        num = 0
        sign = 1
        i = 0
        while i < len(s):
            if s[i] == " ":
                i+=1
            else:
                break
        if i == len(s):
            return 0
        if s[i] == "-":
            sign = -1
            i+=1
        elif s[i] == "+":
            i+=1
        for j in range(i, len(s)):
            if not s[j].isdigit():
                break
            num = num * 10 + int(s[j])
            if num > 2**31-1:
                if sign == -1:
                    return -2**31
                else:
                    return 2**31-1

        return num*sign            
